third_party/wabt/src/interp/jit/jit-backend.cc

```diff
@@ -171,6 +212,173 @@ static void operandToArg(Operand* operand, JITArg& arg) {
 #include "wabt/interp/jit/jit-imath64-inl.h"
 #endif /* SLJIT_32BIT_ARCHITECTURE */
 
+static void floatOperandToArg(sljit_compiler* compiler, Operand* operand, JITArg& arg, unsigned int id) {
+  if (operand->location.type == Operand::Immediate){
+    Instruction* instr = operand->item->asInstruction();
+
+    if ((operand->location.value_info & LocationInfo::kSizeMask) == 1) {
+      arg.argw = static_cast<sljit_s32>(instr->value().value32);
+    } else {
+      arg.argw = static_cast<sljit_sw>(instr->value().value64);
+    }
+
+    sljit_sw floatStorage = id == 0 ? OffsetOfContextField(floatImmediate1) : OffsetOfContextField(floatImmediate2);
+    sljit_emit_op1(compiler, SLJIT_MOV_P, SLJIT_MEM1(kContextReg),
+                   floatStorage , SLJIT_IMM, arg.argw);
+    arg.arg = SLJIT_MEM1(kContextReg);
+    arg.argw = floatStorage;
+  } else {
+    operandToArg(operand, arg);
+  }
+}
+
+static void emitFloatBinary(sljit_compiler* compiler, Instruction* instr) {
+  Operand* operands = instr->operands();
+  JITArg args[3];
+
+  for (unsigned int i = 0; i < 3; ++i) {
+    floatOperandToArg(compiler,operands + i, args[i], i);
+  }
+
+  sljit_s32 opcode;
+  f32Function2Param f32Func = nullptr;
+  f64Function2Param f64Func = nullptr;
+
+  switch (instr->opcode()) {
+    case Opcode::F32Add:
+      opcode = SLJIT_ADD_F32;
+      break;
+    case Opcode::F32Sub:
+      opcode = SLJIT_SUB_F32;
+      break;
+    case Opcode::F32Mul:
+      opcode = SLJIT_MUL_F32;
+      break;
+    case Opcode::F32Div:
+      opcode = SLJIT_DIV_F32;
+      break;
+    case Opcode::F32Max:
+      f32Func = floatMax;
+      break;
+    case Opcode::F32Min:
+      f32Func = floatMin;
+      break;
+    case Opcode::F32Copysign:
+      f32Func = floatCopySign;
+      break;
+    case Opcode::F64Add:
+      opcode = SLJIT_ADD_F64;
+      break;
+    case Opcode::F64Sub:
+      opcode = SLJIT_SUB_F64;
+      break;
+    case Opcode::F64Mul:
+      opcode = SLJIT_MUL_F64;
+      break;
+    case Opcode::F64Div:
+      opcode = SLJIT_DIV_F64;
+      break;
+    case Opcode::F64Max:
+      f64Func = floatMax;
```

Comment:
This could be done with an instruction sequence. This can be done later (maybe a jit instruction if many cpus has instruction for it).